packages/hydrogen-codegen/src/index.ts

```diff
@@ -0,0 +1,139 @@
+import type {Types} from '@graphql-codegen/plugin-helpers';
+import * as addPlugin from '@graphql-codegen/add';
+import * as typescriptOperationPlugin from '@graphql-codegen/typescript-operations';
+import {processSources} from './process-sources';
+import * as dtsPlugin from './dts-plugin';
+
+export type GqlTagConfig = {};
+
+// This comment is used during ESM build:
+//! import {createRequire} from 'module'; const require = createRequire(import.meta.url);
+export const schema = require.resolve(
+  '@shopify/hydrogen-react/storefront.schema.json',
+);
+
+export const preset: Types.OutputPreset<GqlTagConfig> = {
+  buildGeneratesSection: (options) => {
+    const capitalize = (str: string) =>
+      str.charAt(0).toUpperCase() + str.slice(1);
+
+    const sourcesWithOperations = processSources(options.documents, (node) => {
+      if (node.kind === 'FragmentDefinition') {
+        return 'not used';
+      }
+
+      // Match the names generated by typescript-operations plugin:
+      // e.g. 'query Hello {...}' => HelloQuery
+      // -- Anonymous queries are not supported.
+      return capitalize(node.name!.value) + capitalize(node.operation);
+    });
+
+    const sources = sourcesWithOperations.map(({source}) => source);
+
+    const pluginMap = {
+      ...options.pluginMap,
```

Comment:
I'm not too familiar with gql plugins; should we be checking that the user doesn't add the `typescript-operations` and `gen-dts` plugins in addition? I honestly don't know if that would mess things up or not.

Reply:
Yeah, I've added a check before this.